constantine/platforms/abis/nvidia_abi.nim

```diff
@@ -477,14 +485,27 @@ type
 
   CUdevice* = distinct int32
     ## Compute Device handle
-  CUcontext* = distinct pointer
-  CUmodule* = distinct pointer
-  CUfunction* = distinct pointer
-  CUstream* = distinct pointer
-  CUdeviceptr* = distinct pointer
 
-  CUlinkState_st = object
-  CUlinkState* = ptr CUlinkState_st
+  CUcontext* {.importc: "CUcontext", header: "cuda.h".} = distinct pointer
+  CUmodule*  {.importc: "CUmodule", header: "cuda.h".} = distinct pointer
+  CUfunction* {.importc: "CUfunction", header: "cuda.h".} = distinct pointer
+  CUstream* {.importc: "CUstream", header: "cuda.h".} = distinct pointer
+  CUlinkState* {.importc: "CUlinkState", header: "cuda.h".} = distinct pointer # ptr CUlinkState_st
```

Comment:
Do we pass the location of `cuda.h` somewhere or do we assume it's generally in the path.

But in that case shouldn't we use `header: "<cuda.h>"` ?

My worry with this is Windows support, working with headers/dependencies there is just hell and was fuel to rip out BLAS dependencies from Arraymancer: https://github.com/mratsim/Arraymancer/issues/422

Reply:
I don't like the dependency either, but we already had it: https://github.com/mratsim/constantine/blob/master/constantine/platforms/abis/nvidia_abi.nim#L831

(I had to add this due to the issues described in https://github.com/SciNim/nimcuda/issues/27, part of the initial NVRTC PR)